org.civicrm.reqangular/src/common/services/format-currency.service.js

```diff
@@ -2,85 +2,123 @@
      *
-     * @return {promise}
+     * @param  {string|Integer} amount
+     * @param  {object} separators
+     * @return {integer}
      */
-    function fetchSeparators () {
-      promise = promise || getSettings();
+    function removeSeparators (amount, separators) {
+      var expression = configureExpression(separators.decimal);
+      var strippedAmount = amount.toString().replace(new RegExp(expression, 'g'), '');
+      var amountWithDecimal = strippedAmount.replace(separators.decimal, '.');
 
-      return promise.then(function (result) {
-        return {
-          decimal: result.monetaryDecimalPoint,
-          thousand: result.monetaryThousandSeparator
-        };
-      });
+      return amountWithDecimal;
     }
 
     /**
-     * Gets defined settings data form backend
+     * Configure expression for special characters like \ and ]
      *
-     * @param  {Object} params
-     *
-     * @return {Promise}
+     * @param  {string} decimalSeparator
+     * @return {string}
      */
-    function getSettings () {
-      return api.sendGET('Setting', 'get').then(function (data) {
-        return data.values[0];
-      });
+    function configureExpression (decimalSeparator) {
+      var expression;
+
+      // Handle reserved regex characters that might be used as separators
+      switch (decimalSeparator) {
+        case '\\':
+          expression = '[^0-9\\\\]';
+          break;
+        case ']':
+          expression = '[^0-9\\]]';
+          break;
+        default:
+          expression = '[^0-9' + decimalSeparator + ']';
+      }
+
+      return expression;
     }
 
     /**
-     * Removes characters other than decimal separator
-     * Converts the formatted amount to numeric value (making calulable) by
-     * removing the thousand separators and replacing the decimal separator by " ."
+     * Formats the given amount with thousand separators
      *
-     * @param  {string|Integer} amount
-     * @return {integer}
+     * @param  {string|integer} amount
+     * @param  {object} separators
+     * @return {string}
      */
-    function removeCharacters (amount) {
-      return fetchSeparators().then(function (separators) {
-        var expression = '[^0-9' + separators.decimal + ']';
-        var strippedAmount = amount.toString().replace(new RegExp(expression, 'g'), '');
-        var amountWithDecimal = strippedAmount.replace(separators.decimal, '.');
+    function addSeparators (amount, separators) {
+      var amountAfterDecimal; // Keep 2 decimal values after decimal
+      var amountBeforeDecimal;
+      var decimalIndex;
+      var expression;
 
-        return amountWithDecimal;
-      });
+      // Handling special case:
+      // If we are given to format the amount with decimal as "." ,
+      // we need to change "." into defined decimal separator for proper formatting
+      if (+amount && (amount.toString()).indexOf('.') !== -1) {
```

Comment:
I wonder what if I pass '0.0'

Reply:
@bmnepali any feedback on that?

Reply:
yeah! I have already added a test case for this case. it returns '0,0' (where "," is a decimal separator. And final result  after adding  separator will be '0'.